micromeow/fosterfinder
# Feat: Add Email Login & Custom Username and Display Name Selection during Login

## Feat: Add Email Provider and Custom Username/Display Name Selection during OAuth Login

### Description

This pull request adds an email provider to our Next.js application that utilizes NextAuth.js and Prisma DB. It also incorporates the user flow for allowing users to select their preferred username and public display name during OAuth login.

### Changes Made

- [x] Add the email provider integration to NextAuth.js.
- [x] Update the user interface to include an option for email login.
- [x] Implement the custom username and display name selection flow during OAuth login.
- [x] Validate and store the chosen usernames and display names in our Prisma DB.
- [x] Update user profiles with the selected username and display name across the application.

### Steps to test
1. Ensure the application is running locally or on a development environment.
2. Navigate to the login page.
3. Select the email provider option.
4. Enter a valid email address.
5. Follow instructions to verify with a magic link sent to your email address.
6. After successful authentication, you should be prompted to input your preferred username and display name.
7. Enter the desired username and display name and proceed.
8. Verify that the chosen username and display name are recorded in the database, and your display name is correctly displayed in your user profile.
